sm2/src/pke/decrypting.rs

```diff
@@ -1,216 +1,108 @@
-use core::fmt::{self, Debug};
-
-use crate::{
-    AffinePoint, EncodedPoint, FieldBytes, NonZeroScalar, PublicKey, Scalar, SecretKey,
-    arithmetic::field::FieldElement,
-};
-
-use alloc::{borrow::ToOwned, vec::Vec};
+#[cfg(feature = "alloc")]
+use alloc::{vec, vec::Vec};
 use elliptic_curve::{
-    Error, Group, Result,
-    bigint::U256,
-    ops::Reduce,
-    pkcs8::der::Decode,
-    sec1::{FromEncodedPoint, ToEncodedPoint},
-    subtle::{Choice, ConstantTimeEq},
+    CurveArithmetic, Error, Result, SecretKey,
+    sec1::{ModulusSize, ToEncodedPoint},
 };
-use primeorder::PrimeField;
-
-use sm3::{Digest, Sm3, digest::DynDigest};
-
-use super::{Cipher, Mode, encrypting::EncryptingKey, kdf, vec};
-/// Represents a decryption key used for decrypting messages using elliptic curve cryptography.
-#[derive(Clone)]
-pub struct DecryptingKey {
```

Comment:
I'd prefer these names be retained, as they're more consistent with our other crates which identify key types by function